noxfile.py

```diff
@@ -21,51 +32,148 @@
 
 nox.options.error_on_external_run = True
 nox.options.reuse_venv = "yes"
-nox.options.sessions = [
-    "lint",
-    "check-manifest",
-    "slotscheck",
-    "pyright",
-    "test",
-]
 
 PYPROJECT = nox.project.load_toml()
-CI = "CI" in os.environ
+
+SUPPORTED_PYTHONS: Final[List[str]] = nox.project.python_versions(PYPROJECT)
+# TODO(onerandomusername): add 3.14 once CI supports 3.14.
+EXPERIMENTAL_PYTHON_VERSIONS: Final[List[str]] = []
+ALL_PYTHONS: Final[List[str]] = [*SUPPORTED_PYTHONS, *EXPERIMENTAL_PYTHON_VERSIONS]
+MIN_PYTHON: Final[str] = SUPPORTED_PYTHONS[0]
+CI: Final[bool] = "CI" in os.environ
 
 # used to reset cached coverage data once for the first test run only
 reset_coverage = True
 
+if TYPE_CHECKING:
+    ExecutionGroupType = object
+else:
+    ExecutionGroupType = Dict[str, Any]
+
+
+@dataclasses.dataclass
+class ExecutionGroup(ExecutionGroupType):
+    sessions: Tuple[str, ...] = ()
+    python: Optional[str] = None
+    project: bool = True
+    extras: Tuple[str, ...] = ()
+    groups: Tuple[str, ...] = ()
+    dependencies: Tuple[str, ...] = ()
+    experimental: bool = False
+    pyright_paths: Tuple[str, ...] = ()
+
+    def __post_init__(self) -> None:
+        if self.pyright_paths and "pyright" not in self.sessions:
+            self.sessions = (*self.sessions, "pyright")
```

Comment:
Instead of implicitly adding the `pyright` session here, I think it'd be clearer to just raise if it's missing and have it mentioned explicitly in the list below.

Reply:
https://github.com/DisnakeDev/disnake/pull/1341/commits/d1752da97bbaeeafe51eaf0d211ab460ecd6eebf